dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -202,10 +203,31 @@ public Service initializeService(String repositoryId) {
      * @param existingWorkflow
      * @param existingDefaults
      * @param versionName
+     * @param hardRefresh
      * @return workflow with associated workflow versions
      */
     public abstract Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Optional<Workflow> existingWorkflow,
-            Map<String, WorkflowVersion> existingDefaults, Optional<String> versionName);
+            Map<String, WorkflowVersion> existingDefaults, Optional<String> versionName, boolean hardRefresh);
+
+    /**
+     * Determine whether to refresh a version or not
+     * Refresh version if any of the following is true
+     * * this is a hard refresh
+     * * version doesn't exist
+     * * commit id isn't set
+     * * commitId is different
+     * * synced == false
+     *
+     * @param versionName
+     * @param commitId
+     * @param existingDefaults
+     * @param hardRefresh
+     * @return
+     */
+    protected boolean toRefreshVersion(String versionName, String commitId, Map<String, WorkflowVersion> existingDefaults, boolean hardRefresh) {
```

Comment:
Passes in a map just to call get on it, prefer avoid passing the map in to eliminate possibility of side-effects